README
This is a self-contained build environment that creates a statically linked set of RRDtool 1.4 binaries (rrdtool, rrdupdate and rrdcached).

The binaries are built for i386 architecture only, with Mac OS X 10.5 set as the Deployment Target. 

Some minor font and pango configuration is installed into /usr/local/etc but all other libraries are built as static libraries, linked against statically and then removed after the build process. This leaves you with a set of RRDtool binaries that are not dependent on any non-system .dylib library. 

The resulting rrdtool binary is linked as follows:

$ otool -L ~/rrdtool 
/Users/jwilson/rrdtool:
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 111.1.4)
	/usr/lib/libz.1.dylib (compatibility version 1.0.0, current version 1.2.3)
	/System/Library/Frameworks/Carbon.framework/Versions/A/Carbon (compatibility version 2.0.0, current version 136.0.0)
	/usr/lib/libgcc_s.1.dylib (compatibility version 1.0.0, current version 1.0.0)
	/System/Library/Frameworks/CoreServices.framework/Versions/A/CoreServices (compatibility version 1.0.0, current version 32.0.0)
	/System/Library/Frameworks/CoreFoundation.framework/Versions/A/CoreFoundation (compatibility version 150.0.0, current version 476.19.0)
	/System/Library/Frameworks/ApplicationServices.framework/Versions/A/ApplicationServices (compatibility version 1.0.0, current version 34.0.0)

All library dependencies are included in their original form as separate .tar.gz to ensure compatibility.

Xcode is required to build the binaries on Mac OS X. 

This build environment has been tested under Mac OS X 10.6.5 (Snow Leopard) using Xcode 3.2.5

James Wilson
scotchandcode.com

debian/changelog
lcrrdtool (5.0.9) unstable; urgency=low

  * Initial Release.

 -- James Wilson <jwilson@lithiumcorp.com>  Wed, 08 Dec 2010 15:11:49 -0800


build_linux.sh
#!/bin/bash

echo "----------------------------------------------------------------------"
echo "NOTE: This script does use 'sudo make install' to perform install"
echo "      steps that need authentication. You will be prompted to"
echo "      enter your password during the build"
echo ""
echo "      The only thing that is installed outside of this self-contained"
echo "      build are some config files needed by fontconfig and pango. "
echo "      These will go into /usr/local/etc"
echo "----------------------------------------------------------------------"
sleep 4

if [ -f "result/rrdtool" ]; then
  echo "Cheated and used cache!"
  cp result/rrdtool debian/lcrrdtool/usr/bin/lcrrdtool
  cp result/rrdupdate debian/lcrrdtool/usr/bin/lcrrdupdate
  cp result/rrdcached debian/lcrrdtool/usr/bin/lcrrdcached
  exit 0
fi

PKGCONFIG=pkg-config-0.25
LIBICONV=libiconv-1.13.1
LIBPNG=libpng-1.4.4
PIXMAN=pixman-0.19.6
FONTCONFIG=fontconfig-2.8.0
POPPLER=poppler-0.14.4
CAIRO=cairo-1.10.0
EXPAT=expat-2.0.1
GETTEXT=gettext-0.18.1.1
GLIB=glib-2.26.0
PANGO=pango-1.28.3
RRDTOOL=rrdtool-1.4.4
LIBXML=libxml2-2.7.7
FREETYPE=freetype-2.4.3

STAGE=$PWD/stage

PKG_CONFIG=$STAGE/bin/pkg-config
PKG_CONFIG_PATH=$STAGE/lib/pkgconfig  
PATH=$PATH:$STAGE/bin
CFLAGS=""
LDFLAGS="-L$STAGE/lib" 

CONFIGURE_FLAGS="--prefix=$STAGE --enable-static --sysconfdir=/usr/etc"

#
# Staging
#

rm -rf $STAGE
mkdir -p $STAGE

# 
# Build 
#

# pkg-config
tar zxvf $PKGCONFIG.tar.gz
cd $PKGCONFIG
CFLAGS=$CFLAGS \
CXXFLAGS=$CFLAGS \
LDFLAGS=$LDFLAGS \
PKG_CONFIG=$PKG_CONFIG \
PKG_CONFIG_PATH=$PKG_CONFIG_PATH \
./configure $CONFIGURE_FLAGS && make && echo "Using sudo to install, you may be asked for your password" && sudo make install
if [ $? -ne 0 ]; then
  exit 1
fi
cd ..
rm -rf $PKGCONFIG

# libiconv
tar zxvf $LIBICONV.tar.gz
cd $LIBICONV
CFLAGS=$CFLAGS \
CXXFLAGS=$CFLAGS \
LDFLAGS=$LDFLAGS \
PKG_CONFIG=$PKG_CONFIG \
PKG_CONFIG_PATH=$PKG_CONFIG_PATH \
./configure $CONFIGURE_FLAGS && make && echo "Using sudo to install, you may be asked for your password" && sudo make install
if [ $? -ne 0 ]; then
  exit 1
fi
cd ..
rm -rf $LIBICONV

# libpng
tar zxvf $LIBPNG.tar.gz
cd $LIBPNG
CFLAGS=$CFLAGS \
CXXFLAGS=$CFLAGS \
LDFLAGS=$LDFLAGS \
PKG_CONFIG=$PKG_CONFIG \
PKG_CONFIG_PATH=$PKG_CONFIG_PATH \
./configure $CONFIGURE_FLAGS && make && echo "Using sudo to install, you may be asked for your password" && sudo make install
if [ $? -ne 0 ]; then
  exit 1
fi
cd ..
rm -rf $LIBPNG

# pixman
tar zxvf $PIXMAN.tar.gz
cd $PIXMAN
CFLAGS=$CFLAGS \
CXXFLAGS=$CFLAGS \
LDFLAGS=$LDFLAGS \
PKG_CONFIG=$PKG_CONFIG \
PKG_CONFIG_PATH=$PKG_CONFIG_PATH \
./configure $CONFIGURE_FLAGS && make && echo "Using sudo to install, you may be asked for your password" && sudo make install
if [ $? -ne 0 ]; then
  exit 1
fi
cd ..
rm -rf $PIXMAN

# freetype
tar zxvf $FREETYPE.tar.gz
cd $FREETYPE
CFLAGS=$CFLAGS \
CXXFLAGS=$CFLAGS \
LDFLAGS=$LDFLAGS \
PKG_CONFIG=$PKG_CONFIG \
PKG_CONFIG_PATH=$PKG_CONFIG_PATH \
./configure --prefix=$STAGE --enable-static && make && echo "Using sudo to install, you may be asked for your password" && sudo make install
if [ $? -ne 0 ]; then
  exit 1
fi
cd ..
rm -rf $FREETYPE

# fontconfig
tar zxvf $FONTCONFIG.tar.gz
cd $FONTCONFIG
sed -i '' -e 's/use_iconv=1/use_iconv=0/g' configure.in
libtoolize --force
aclocal
autoconf
automake
CFLAGS=$CFLAGS \
CXXFLAGS=$CFLAGS \
LDFLAGS="$LDFLAGS" \
PKG_CONFIG=$PKG_CONFIG \
PKG_CONFIG_PATH=$PKG_CONFIG_PATH \
LIBXML2_CFLAGS="-I/usr/include/libxml2" \
LIBXML2_LIBS="-lxml2" \
./configure $CONFIGURE_FLAGS --with-freetype-config=$STAGE/bin/freetype-config && make && echo "Using sudo to install, you may be asked for your password" && sudo make install
if [ $? -ne 0 ]; then
  exit 1
fi
cd ..
rm -rf $FONTCONFIG

# poppler
tar zxvf $POPPLER.tar.gz
cd $POPPLER
CFLAGS=$CFLAGS \
CXXFLAGS=$CFLAGS \
LDFLAGS=$LDFLAGS \
PKG_CONFIG=$PKG_CONFIG \
PKG_CONFIG_PATH=$PKG_CONFIG_PATH \
./configure $CONFIGURE_FLAGS --disable-poppler-qt --disable-poppler-cpp --disable-poppler-qt4 --enable-xpdf-headers --enable-zlib --disable-gtk-test --without-x
if [ $? -ne 0 ]; then
  exit 1
fi
#sed -i '' -e "s/DEFAULT_INCLUDES \=/DEFAULT_INCLUDES \= -I..\/..\/stage\/include/" cpp/Makefile
#if [ $? -ne 0 ]; then
#  exit 1
#fi
make && echo "Using sudo to install, you may be asked for your password" && sudo make install
if [ $? -ne 0 ]; then
  exit 1
fi
cd ..
rm -rf $POPPLER

# cairo
tar zxvf $CAIRO.tar.gz
cd $CAIRO
CFLAGS=$CFLAGS \
CXXFLAGS=$CFLAGS \
LDFLAGS=$LDFLAGS \
PKG_CONFIG=$PKG_CONFIG \
PKG_CONFIG_PATH=$PKG_CONFIG_PATH \
./configure $CONFIGURE_FLAGS --enable-static --enable-xlib=no --enable-xlib-render=no --enable-win32=no && make && echo "Using sudo to install, you may be asked for your password" && sudo make install
if [ $? -ne 0 ]; then
  exit 1
fi
cd ..
rm -rf $CAIRO

# expat
tar zxvf $EXPAT.tar.gz
cd $EXPAT
CFLAGS=$CFLAGS \
CXXFLAGS=$CFLAGS \
LDFLAGS=$LDFLAGS \
PKG_CONFIG=$PKG_CONFIG \
PKG_CONFIG_PATH=$PKG_CONFIG_PATH \
./configure $CONFIGURE_FLAGS && make && echo "Using sudo to install, you may be asked for your password" && sudo make install
if [ $? -ne 0 ]; then
  exit 1
fi
cd ..
rm -rf $EXPAT

# gettext
tar zxvf $GETTEXT.tar.gz
cd $GETTEXT
CFLAGS=$CFLAGS \
CXXFLAGS=$CFLAGS \
LDFLAGS=$LDFLAGS \
PKG_CONFIG=$PKG_CONFIG \
PKG_CONFIG_PATH=$PKG_CONFIG_PATH \
./configure $CONFIGURE_FLAGS && make && echo "Using sudo to install, you may be asked for your password" && sudo make install
if [ $? -ne 0 ]; then
  exit 1
fi
cd ..
rm -rf $GETTEXT

# libxml
tar zxvf $LIBXML.tar.gz
cd $LIBXML
CFLAGS=$CFLAGS \
CXXFLAGS=$CFLAGS \
LDFLAGS=$LDFLAGS \
PKG_CONFIG=$PKG_CONFIG \
PKG_CONFIG_PATH=$PKG_CONFIG_PATH \
./configure $CONFIGURE_FLAGS && make && echo "Using sudo to install, you may be asked for your password" && sudo make install
if [ $? -ne 0 ]; then
  exit 1
fi
cd ..
rm -rf $LIBXML

# glib
tar zxvf $GLIB.tar.gz
cd $GLIB
CFLAGS=$CFLAGS \
CXXFLAGS=$CFLAGS \
LDFLAGS=$LDFLAGS \
PKG_CONFIG=$PKG_CONFIG \
PKG_CONFIG_PATH=$PKG_CONFIG_PATH \
./configure $CONFIGURE_FLAGS && make && echo "Using sudo to install, you may be asked for your password" && sudo make install
if [ $? -ne 0 ]; then
  exit 1
fi
cd ..
rm -rf $GLIB

# pango
tar zxvf $PANGO.tar.gz
cd $PANGO
sed -i '' -e 's/have_atsui=true/have_atsui=false/g' configure
CFLAGS=$CFLAGS \
CXXFLAGS=$CFLAGS \
LDFLAGS=$LDFLAGS \
PKG_CONFIG=$PKG_CONFIG \
PKG_CONFIG_PATH=$PKG_CONFIG_PATH \
./configure $CONFIGURE_FLAGS --with-dynamic-modules=no --with-included-modules=basic-fc --enable-static --without-x && make && echo "Using sudo to install, you may be asked for your password" && sudo make install
if [ $? -ne 0 ]; then
  exit 1
fi
cd ..
rm -rf $PANGO

# rrdtool
rm -rf $RRDTOOL
tar zxvf $RRDTOOL.tar.gz
cd $RRDTOOL
CFLAGS=$CFLAGS \
CXXFLAGS=$CFLAGS \
LDFLAGS="$LDFLAGS -lpthread" \
PKG_CONFIG=$PKG_CONFIG \
PKG_CONFIG_PATH=$PKG_CONFIG_PATH \
./configure $CONFIGURE_FLAGS --disable-rrdcgi --enable-static-programs --disable-pthread\
  --disable-libdbi --disable-perl --disable-ruby --disable-lua --disable-tcl \
  PKGCONFIG=$PKG_CONFIG && make && echo "Using sudo to install, you may be asked for your password" && sudo make install
if [ $? -ne 0 ]; then
  exit 1
fi
cd ..
rm -rf $RRDTOOL

#
# Get Result
#

rm -rf result
mkdir -p result
cp $STAGE/bin/rrd* result
mkdir -p debian/lcrrdtool/usr/bin
cp $STAGE/bin/rrdtool debian/lcrrdtool/usr/bin/lcrrdtool
cp $STAGE/bin/rrdupdate debian/lcrrdtool/usr/bin/lcrrdupdate
cp $STAGE/bin/rrdcached debian/lcrrdtool/usr/bin/lcrrdcached

echo "Cleaning up the staging area, you may be asked for your password."
sudo rm -rf $STAGE

echo "----------------------------------------------------------------------"
echo "The static build of RRDtool has completed and the rrdtool, rrdupdate"
echo "and rrdcached binaries can be found in the 'result' directory"
echo "----------------------------------------------------------------------"


build.sh
#!/bin/sh

echo "----------------------------------------------------------------------"
echo "NOTE: This script does use 'sudo make install' to perform install"
echo "      steps that need authentication. You will be prompted to"
echo "      enter your password during the build"
echo ""
echo "      The only thing that is installed outside of this self-contained"
echo "      build are some config files needed by fontconfig and pango. "
echo "      These will go into /usr/local/etc"
echo "----------------------------------------------------------------------"
sleep 4

PKGCONFIG=pkg-config-0.25
LIBICONV=libiconv-1.13.1
LIBPNG=libpng-1.4.4
PIXMAN=pixman-0.19.6
FONTCONFIG=fontconfig-2.8.0
POPPLER=poppler-0.14.4
CAIRO=cairo-1.10.0
EXPAT=expat-2.0.1
GETTEXT=gettext-0.18.1.1
GLIB=glib-2.26.0
PANGO=pango-1.28.3
RRDTOOL=rrdtool-1.4.4
LIBXML=libxml2-2.7.7
FREETYPE=freetype-2.4.3

STAGE=$PWD/stage

PKG_CONFIG=$STAGE/bin/pkg-config
PKG_CONFIG_PATH=$STAGE/lib/pkgconfig  
PATH=$PATH:$STAGE/bin
CFLAGS="-mmacosx-version-min=10.5 -isysroot /Developer/SDKs/MacOSX10.5.sdk -Wl,-search_paths_first -O -arch i386 -I$STAGE/include"
LDFLAGS="-L$STAGE/lib -arch i386" 

CONFIGURE_FLAGS="--disable-dependency-tracking --prefix=$STAGE --enable-static --sysconfdir=/usr/local/etc"

#
# Staging
#

rm -rf $STAGE
mkdir -p $STAGE

# 
# Build 
#

# pkg-config
tar zxvf $PKGCONFIG.tar.gz
cd $PKGCONFIG
CFLAGS=$CFLAGS \
CXXFLAGS=$CFLAGS \
LDFLAGS=$LDFLAGS \
PKG_CONFIG=$PKG_CONFIG \
PKG_CONFIG_PATH=$PKG_CONFIG_PATH \
./configure $CONFIGURE_FLAGS && make && echo "Using sudo to install, you may be asked for your password" && sudo make install
if [ $? -ne 0 ]; then
  exit 1
fi
cd ..
rm -rf $PKGCONFIG

# libiconv
tar zxvf $LIBICONV.tar.gz
cd $LIBICONV
CFLAGS=$CFLAGS \
CXXFLAGS=$CFLAGS \
LDFLAGS=$LDFLAGS \
PKG_CONFIG=$PKG_CONFIG \
PKG_CONFIG_PATH=$PKG_CONFIG_PATH \
./configure $CONFIGURE_FLAGS && make && echo "Using sudo to install, you may be asked for your password" && sudo make install
if [ $? -ne 0 ]; then
  exit 1
fi
cd ..
rm -rf $LIBICONV

# libpng
tar zxvf $LIBPNG.tar.gz
cd $LIBPNG
CFLAGS=$CFLAGS \
CXXFLAGS=$CFLAGS \
LDFLAGS=$LDFLAGS \
PKG_CONFIG=$PKG_CONFIG \
PKG_CONFIG_PATH=$PKG_CONFIG_PATH \
./configure $CONFIGURE_FLAGS && make && echo "Using sudo to install, you may be asked for your password" && sudo make install
if [ $? -ne 0 ]; then
  exit 1
fi
cd ..
rm -rf $LIBPNG

# pixman
tar zxvf $PIXMAN.tar.gz
cd $PIXMAN
CFLAGS=$CFLAGS \
CXXFLAGS=$CFLAGS \
LDFLAGS=$LDFLAGS \
PKG_CONFIG=$PKG_CONFIG \
PKG_CONFIG_PATH=$PKG_CONFIG_PATH \
./configure $CONFIGURE_FLAGS && make && echo "Using sudo to install, you may be asked for your password" && sudo make install
if [ $? -ne 0 ]; then
  exit 1
fi
cd ..
rm -rf $PIXMAN

# freetype
tar zxvf $FREETYPE.tar.gz
cd $FREETYPE
CFLAGS=$CFLAGS \
CXXFLAGS=$CFLAGS \
LDFLAGS=$LDFLAGS \
PKG_CONFIG=$PKG_CONFIG \
PKG_CONFIG_PATH=$PKG_CONFIG_PATH \
./configure --prefix=$STAGE --enable-static && make && echo "Using sudo to install, you may be asked for your password" && sudo make install
if [ $? -ne 0 ]; then
  exit 1
fi
cd ..
rm -rf $FREETYPE

# fontconfig
tar zxvf $FONTCONFIG.tar.gz
cd $FONTCONFIG
CFLAGS=$CFLAGS \
CXXFLAGS=$CFLAGS \
LDFLAGS=$LDFLAGS \
PKG_CONFIG=$PKG_CONFIG \
PKG_CONFIG_PATH=$PKG_CONFIG_PATH \
./configure $CONFIGURE_FLAGS --with-add-fonts=/System/Library/Fonts --with-freetype-config=$STAGE/bin/freetype-config && make && echo "Using sudo to install, you may be asked for your password" && sudo make install
if [ $? -ne 0 ]; then
  exit 1
fi
cd ..
rm -rf $FONTCONFIG

# poppler
tar zxvf $POPPLER.tar.gz
cd $POPPLER
CFLAGS=$CFLAGS \
CXXFLAGS=$CFLAGS \
LDFLAGS=$LDFLAGS \
PKG_CONFIG=$PKG_CONFIG \
PKG_CONFIG_PATH=$PKG_CONFIG_PATH \
./configure $CONFIGURE_FLAGS --disable-poppler-qt --disable-poppler-qt4 --enable-xpdf-headers --enable-zlib --disable-gtk-test --without-x
if [ $? -ne 0 ]; then
  exit 1
fi
sed -i '' -e "s/DEFAULT_INCLUDES \=/DEFAULT_INCLUDES \= -I..\/..\/stage\/include/" cpp/Makefile
if [ $? -ne 0 ]; then
  exit 1
fi
make && echo "Using sudo to install, you may be asked for your password" && sudo make install
if [ $? -ne 0 ]; then
  exit 1
fi
cd ..
rm -rf $POPPLER

# cairo
tar zxvf $CAIRO.tar.gz
cd $CAIRO
CFLAGS=$CFLAGS \
CXXFLAGS=$CFLAGS \
LDFLAGS=$LDFLAGS \
PKG_CONFIG=$PKG_CONFIG \
PKG_CONFIG_PATH=$PKG_CONFIG_PATH \
./configure $CONFIGURE_FLAGS --enable-static --enable-xlib=no --enable-xlib-render=no --enable-win32=no && make && echo "Using sudo to install, you may be asked for your password" && sudo make install
if [ $? -ne 0 ]; then
  exit 1
fi
cd ..
rm -rf $CAIRO

# expat
tar zxvf $EXPAT.tar.gz
cd $EXPAT
CFLAGS=$CFLAGS \
CXXFLAGS=$CFLAGS \
LDFLAGS=$LDFLAGS \
PKG_CONFIG=$PKG_CONFIG \
PKG_CONFIG_PATH=$PKG_CONFIG_PATH \
./configure $CONFIGURE_FLAGS && make && echo "Using sudo to install, you may be asked for your password" && sudo make install
if [ $? -ne 0 ]; then
  exit 1
fi
cd ..
rm -rf $EXPAT

# gettext
tar zxvf $GETTEXT.tar.gz
cd $GETTEXT
CFLAGS=$CFLAGS \
CXXFLAGS=$CFLAGS \
LDFLAGS=$LDFLAGS \
PKG_CONFIG=$PKG_CONFIG \
PKG_CONFIG_PATH=$PKG_CONFIG_PATH \
./configure $CONFIGURE_FLAGS && make && echo "Using sudo to install, you may be asked for your password" && sudo make install
if [ $? -ne 0 ]; then
  exit 1
fi
cd ..
rm -rf $GETTEXT

# libxml
tar zxvf $LIBXML.tar.gz
cd $LIBXML
CFLAGS=$CFLAGS \
CXXFLAGS=$CFLAGS \
LDFLAGS=$LDFLAGS \
PKG_CONFIG=$PKG_CONFIG \
PKG_CONFIG_PATH=$PKG_CONFIG_PATH \
./configure $CONFIGURE_FLAGS && make && echo "Using sudo to install, you may be asked for your password" && sudo make install
if [ $? -ne 0 ]; then
  exit 1
fi
cd ..
rm -rf $LIBXML

# glib
tar zxvf $GLIB.tar.gz
cd $GLIB
CFLAGS=$CFLAGS \
CXXFLAGS=$CFLAGS \
LDFLAGS=$LDFLAGS \
PKG_CONFIG=$PKG_CONFIG \
PKG_CONFIG_PATH=$PKG_CONFIG_PATH \
./configure $CONFIGURE_FLAGS && make && echo "Using sudo to install, you may be asked for your password" && sudo make install
if [ $? -ne 0 ]; then
  exit 1
fi
cd ..
rm -rf $GLIB

# pango
tar zxvf $PANGO.tar.gz
cd $PANGO
sed -i '' -e 's/have_atsui=true/have_atsui=false/g' configure
CFLAGS=$CFLAGS \
CXXFLAGS=$CFLAGS \
LDFLAGS=$LDFLAGS \
PKG_CONFIG=$PKG_CONFIG \
PKG_CONFIG_PATH=$PKG_CONFIG_PATH \
./configure $CONFIGURE_FLAGS --with-dynamic-modules=no --with-included-modules=basic-fc --enable-static --without-x && make && echo "Using sudo to install, you may be asked for your password" && sudo make install
if [ $? -ne 0 ]; then
  exit 1
fi
cd ..
rm -rf $PANGO

# rrdtool
tar zxvf $RRDTOOL.tar.gz
cd $RRDTOOL
CFLAGS=$CFLAGS \
CXXFLAGS=$CFLAGS \
LDFLAGS=$LDFLAGS \
PKG_CONFIG=$PKG_CONFIG \
PKG_CONFIG_PATH=$PKG_CONFIG_PATH \
./configure $CONFIGURE_FLAGS --disable-rrdcgi --enable-static-programs \
  --disable-libdbi --disable-perl --disable-ruby --disable-lua --disable-tcl\
  PKGCONFIG=$PKG_CONFIG && make && echo "Using sudo to install, you may be asked for your password" && sudo make install
if [ $? -ne 0 ]; then
  exit 1
fi
cd ..
rm -rf $RRDTOOL

#
# Get Result
#

rm -rf result
mkdir -p result
cp $STAGE/bin/rrd* result

echo "Cleaning up the staging area, you may be asked for your password."
sudo rm -rf $STAGE

echo "----------------------------------------------------------------------"
echo "The static build of RRDtool has completed and the rrdtool, rrdupdate"
echo "and rrdcached binaries can be found in the 'result' directory"
echo "----------------------------------------------------------------------"
